narwhal/primary/src/state_handler.rs

```diff
@@ -145,7 +149,7 @@ impl StateHandler {
 
                 Some(message) = self.rx_state_handler.recv() => {
                     // Notify our workers
-                    self.notify_our_workers(message.to_owned());
+                    let notify_handlers = self.notify_our_workers(message.to_owned());
```

Comment:
Do we know what the semantics around this are supposed to be? is a successful response mean the worker has properly shutdown? How does that work if the network we're sending this message on is intended to be shutdown as well?

Reply:
To be honest this is a quite fragile as logic to begin with and that's why we want to remove the call anyways (FYI https://mysten.atlassian.net/browse/SUI-984) and instead follow an approach where we'll manage the primary/worker shutdown independently as part of the upcoming iterations. So I don't feel confidence on the approach in either case (reliable vs unreliable send).

Now , for the current semantics I expect for the worker to manage to respond before shutting down - but I want to clarify that we don't have strong guarantees here. Maybe we could do some very hackish stuff in the worker to delay the propagation of the reconfiguration signal by some time, so we can get more confidence that the endpoint will manage to return .

Reply:
Yeah we can't communicate this signal over an interface that is going to be shutdown, but it sounds like you already have plans to address that